src/logic/hooks/useEstimateSafeTxGas.tsx

```diff
@@ -22,30 +23,25 @@ export const useEstimateSafeTxGas = ({
   txAmount,
   operation,
 }: UseEstimateSafeTxGasProps): string => {
-  const [safeTxGasEstimation, setSafeTxGasEstimation] = useState<string>('0')
+  const defaultEstimation = '0'
   const { address: safeAddress, currentVersion: safeVersion } = useSelector(currentSafe) ?? {}
 
-  useEffect(() => {
-    if (!isCreation || isRejectTx) return
-    const estimateSafeTxGasCall = async () => {
-      try {
-        const safeTxGasEstimation = await estimateSafeTxGas(
-          {
-            safeAddress,
-            txData,
-            txRecipient,
-            txAmount: txAmount || '0',
-            operation: operation || Operation.CALL,
-          },
-          safeVersion,
-        )
-        setSafeTxGasEstimation(safeTxGasEstimation)
-      } catch (error) {
-        console.warn(error.message)
-      }
-    }
-    estimateSafeTxGasCall()
+  const requestSafeTxGas = useCallback((): Promise<string> => {
+    if (!isCreation || isRejectTx) return Promise.resolve(defaultEstimation)
+
+    return estimateSafeTxGas(
+      {
+        safeAddress,
+        txData,
+        txRecipient,
+        txAmount: txAmount || '0',
+        operation: operation || Operation.CALL,
+      },
+      safeVersion,
+    )
   }, [isCreation, isRejectTx, operation, safeAddress, safeVersion, txAmount, txData, txRecipient])
 
-  return safeTxGasEstimation
+  const { result } = useAsync<string>(requestSafeTxGas)
```

Comment:
No error handling?

Reply:
Just checked, `estimateSafeTxGas` has a catch inside, so it won't throw.

Reply:
Otherwise one would do this:
```
useEffect(() => {
  error && logError(Errors.xxx, error.message)
}, [ error ])
```

Reply:
Can you add/extend the existing test for it?